scripts/dts/gen_defines.py

```diff
@@ -195,23 +253,69 @@ def relativize(path):
 def write_regs(node):
     # Writes address/size output for the registers in the node's 'reg' property
 
-    def write_reg(reg, base_ident, val):
-        # Drop '_0' from the identifier if there's a single register, for
-        # backwards compatibility
-        if len(reg.node.regs) > 1:
-            ident = f"{base_ident}_{reg.node.regs.index(reg)}"
+    if not node.regs:
+        return
+
+    primary_addrs = {}
+    primary_sizes = {}
+
+    def write_regs_for_ident(ident):
+        # Write BASE_ADDRESS and SIZE macros for a given identifier
+        # 'ident'. If we have already generated primary address and
+        # size macros and saved in them in primary_addrs and
+        # primary_sizes, we just reuse those. Otherwise (i.e. the
+        # first time this is called), they are generated from the
+        # actual reg.addr and reg.size attributes, and the names of
+        # the primary macros are saved.
+
+        for reg_i, reg in enumerate(node.regs):
+            idx = f"_{reg_i}"
+            name = f"{str2ident(reg.name)}_" if reg.name else ""
```

Comment:
Wonder if this is redundant. `name` is only used within an `if reg.name` below, so it seems it'd never be `""` if it's used.

Maybe `f"{str2ident(reg.name)}_"` could be passed directly to `write_reg()` instead.